### ISSUE TITLE
Support schema override

### ISSUE THREAD

**Issue by rschweter** [bali182/autocomplete-json#50]
There are some entries in http://schemastore.org/api/json/catalog.json that don't have a fileMatch attribute. It would be nice if we could specify a schema for a specific filename to help support these.

**Comment by rschweter** [comment]
Later saw the wiki writeup on a custom schema, did that, but then realized that autocomplete isn't working for any json (e.g. package.json). New to atom packages so I'm digging around to see if I can provide any more info for you. Running 1.7.4 on OSX.

**Comment by bali182** [comment]
Yes, this also bothers me. To be able to specify the schema for a file without `fileMatch` we would need a way to specify the filename-schema binding on plugin level, which I think would require some UI too, and I'm also not very familiar on how to create UI-s in Atom.

Another solution would be to update schemastore. Is the schema you are using have a fixed file name that is just missing from the store? 

Do you have the plugin on github somewhere? I gladly take a look & get back to you on how to fix it :) It is also possible, that you found a bug somewhere.

**Comment by rschweter** [comment]
Nothing fancy, just 

function SwaggerSchemaProvider() {
    return {
        getSchemaURI: function() {
            return 'http://json.schemastore.org/swagger-2.0';
        },

        getFilePattern(): function() {
            return 'api.json';
        }
    };
}

module.exports = {
    SwaggerSchemaProvider: SwaggerSchemaProvider;
};

Which results in a "schemas not available" error on every ctrl-space.  I'll put it up on github later when i get a chance to dig in deeper.

**Comment by bali182** [comment]
Hmm, this looks good. Just to be sure, do you have this in your package.json? 

```
{
  "main": "path/to/SwaggerSchemaProvider.js",
  "providedServices": {
    "jsonschema.provider": {
      "versions": {
        "1.0.0": "SwaggerSchemaProvider"
      }
    }
  }
}
```

If you do, you probably found a bug, I'll look at it in the evening.

**Comment by bali182** [comment]
Found the issue! The schema you want to use refers to another schema (other than itself):

```json
...
"title": {
  "$ref": "http://json-schema.org/draft-04/schema#/properties/title"
}
...
```

This is currently not supported, but PRs are welcome. If I have the time, I'll do it, but I can't promise anything.

**Comment by bali182** [comment]
@rschweter  I implemented this in 5.3.0. If you are still using this would you mind giving it a try? I simply created a `swagger.json` file and the schema seemed to work properly.
